platform/android/MapboxGLAndroidSDK/src/main/java/com/mapbox/mapboxsdk/style/expressions/Expression.java

```diff
@@ -0,0 +1,1462 @@
+   * If the input is a boolean, the result is true or false.
+   * If the input is a number, it is converted to a string by NumberToString in the ECMAScript Language Specification.
+   * If the input is a color, it is converted to a string of the form "rgba(r,g,b,a)",
+   * where `r`, `g`, and `b` are numerals ranging from 0 to 255, and `a` ranges from 0 to 1.
+   * Otherwise, the input is converted to a string in the format specified by the JSON.stringify in the ECMAScript
+   * Language Specification.
+   *
+   * @param expressions varargs to provide expressions and values
+   * @return expression
+   */
+  public static Expression toString(Object... expressions) {
+    return new Expression("to-string", expressions);
+  }
+
+  /**
+   * Converts the input value to a number, if possible.
+   * If the input is null or false, the result is 0.
+   * If the input is true, the result is 1.
+   * If the input is a string, it is converted to a number as specified by the ECMAScript Language Specification.
+   * If multiple values are provided, each one is evaluated in order until the first successful conversion is obtained.
+   * If none of the inputs can be converted, the expression is an error.
+   *
+   * @param expressions varargs to provide expressions and values
+   * @return expression
+   */
+  public static Expression toNumber(Object... expressions) {
+    return new Expression("to-number", expressions);
+  }
+
+  /**
+   * "Converts the input value to a boolean. The result is `false` when then input is an empty string, 0, false,
+   * null, or NaN; otherwise it is true.
+   *
+   * @param expressions varargs to provide expressions and values
+   * @return expression
+   */
+  public static Expression toBool(Object... expressions) {
+    return new Expression("to-boolean", expressions);
+  }
+
+  /**
+   * Converts the input value to a color. If multiple values are provided,
+   * each one is evaluated in order until the first successful conversion is obtained.
+   * If none of the inputs can be converted, the expression is an error.
+   *
+   * @param expressions varargs to provide expressions and values
+   * @return expression
+   */
+  public static Expression toColor(Object... expressions) {
+    return new Expression("to-color", expressions);
+  }
+
+  //
+  // Variable binding
+  //
+
+  /**
+   * Binds expressions to named variables,
+   * which can then be referenced in the result expression using {@link #var(String)} or {@link #var(Object...)}.
+   *
+   * @param expressions varargs to provide expressions and values
+   * @return expression
+   */
+  public static Expression let(Object... expressions) {
+    return new Expression("let", expressions);
+  }
+
+  /**
+   * References variable bound using let.
+   *
+   * @param expressions varargs to provide expressions and values
+   * @return expression
+   */
+  public static Expression var(Object... expressions) {
+    return new Expression("var", expressions);
+  }
```

Comment:
The variable name argument must be a string literal -- it can't be a subexpression -- so it might make sense to remove this overload.

Reply:
can the string literal come from another expression? 

eg.
  var(get("featureKey"))

If that is the case I will remove the varargs approach and expose the following:
 - var(Expression expression) 
 - var(String string)

Reply:
Nope -- "literal" here means it can't be the output of an expression, but has to be a string directly.